biz.aQute.bndlib/src/aQute/bnd/header/Attrs.java

```diff
@@ -614,4 +615,24 @@ public Attrs with(String key, String value) {
 		put(key, value);
 		return this;
 	}
+
+	/**
+	 * Return a new Attrs that has only the attributes that match the predicate.
+	 * The primary use case for this is AttributeClasses.
+	 *
+	 * @param predicate a predicate that returns true if the attribute must be
+	 *            included in the result
+	 * @return a new Attrs that can be used and modified by the caller
+	 */
+
+	public Attrs select(Predicate<String> predicate) {
+		Attrs attrs = new Attrs();
+		forEach((k, v) -> {
+			if (predicate.test(k)) {
+				attrs.map.put(k, v);
+				attrs.types.put(k, this.types.get(k));
```

Comment:
```suggestion
				attrs.put(k, getType(k), v);
```

This uses the new put method which take a type.